### PR TITLE
Bug 2093527: bump to kube 1.24.0

### PR DESCRIPTION
**Pull request by sanchezl**
* Upgrade to go 1.18
* Bump `k8s.io/*` to `v1.24.0` 
* Bump `github.com/openshift/*` to `release-4.11`
* resource names must be lowercase only during registration.
* fix TestOpenAPIRoundtrip to handle legacy schema.
* PodAffinityNamespaceSelector is GA.
* error wrapping directive not supported is now an error.
* Fix referenced to deprecated clustername property
* Use alternative for removed `rest.ValidNamespace`

### CODE REVIEW

**Comment by sanchezl** [comment]
/hold

Needs and updated openshift/kubernetes-apiserver

File: go.mod [hunk 1]
```diff
@@ -16,30 +16,30 @@ require (
 	github.com/hashicorp/golang-lru v0.5.1
 	github.com/jteeuwen/go-bindata v3.0.8-0.20151023091102-a0ff2567cfb7+incompatible
 	github.com/opencontainers/go-digest v1.0.0
-	github.com/opencontainers/image-spec v1.0.2-0.20190823105129-775207bd45b6
+	github.com/opencontainers/image-spec v1.0.2
```
Comment (soltysh) [inline go.mod:19]: I'd also expect openshift libs being updated with it

File: pkg/apps/apiserver/apiserver.go [hunk 2]
```diff
@@ -123,11 +123,11 @@ func (c *completedConfig) newV1RESTStorage() (map[string]rest.Storage, error) {
 	deployConfigRollbackStorage := deployrollback.NewREST(openshiftAppsClient, kubeClient)
 
 	v1Storage := map[string]rest.Storage{}
-	v1Storage["deploymentConfigs"] = deployConfigStorage
-	v1Storage["deploymentConfigs/scale"] = deployConfigScaleStorage
-	v1Storage["deploymentConfigs/status"] = deployConfigStatusStorage
-	v1Storage["deploymentConfigs/rollback"] = deployConfigRollbackStorage
-	v1Storage["deploymentConfigs/log"] = deploylogregistry.NewREST(openshiftAppsClient.AppsV1(), kubeClient)
-	v1Storage["deploymentConfigs/instantiate"] = dcInstantiateStorage
+	v1Storage["deploymentconfigs"] = deployConfigStorage
```
Comment (soltysh) [inline pkg/apps/apiserver/apiserver.go:126]: Why this change, that's rather unexpected?

Comment (sanchezl) [reply pkg/apps/apiserver/apiserver.go:126]: See failures in past test runs.

File: go.mod (hunk 1 above)
Comment (sanchezl) [reply go.mod:19]: Done.

File: pkg/apps/apiserver/apiserver.go (hunk 2 above)
Comment (sanchezl) [reply pkg/apps/apiserver/apiserver.go:126]: > Why this change, that's rather unexpected?

Validation started enforcing that resource names should be lowercase in https://github.com/kubernetes/kubernetes/pull/108263.

**Review by soltysh — APPROVED** [review]
/approve

**Comment by sanchezl** [comment]
Superseded by https://github.com/openshift/openshift-apiserver/pull/320.
